nixos/modules/services/misc/prowlarr.nix

```diff
@@ -0,0 +1,73 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.prowlarr;
+
+in
+{
+  options = {
+    services.prowlarr = {
+      enable = mkEnableOption "Prowlarr";
+
+      dataDir = mkOption {
+        type = types.str;
+        default = "/var/lib/prowlarr/.config/Prowlarr";
+        description = "The directory where Prowlarr stores its data files.";
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Open ports in the firewall for the Prowlarr web interface.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "prowlarr";
+        description = "User account under which Prowlarr runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "prowlarr";
+        description = "Group under which Prowlarr runs.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.tmpfiles.rules = [
+      "d '${cfg.dataDir}' 0700 ${cfg.user} ${cfg.group} - -"
+    ];
+
+    systemd.services.prowlarr = {
+      description = "Prowlarr";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        Type = "simple";
+        User = cfg.user;
+        Group = cfg.group;
+        ExecStart = "${pkgs.prowlarr}/bin/Prowlarr -nobrowser -data='${cfg.dataDir}'";
+        Restart = "on-failure";
+      };
+    };
+
+    networking.firewall = mkIf cfg.openFirewall {
+      allowedTCPPorts = [ 9696 ];
```

Comment:
Is the port not configurable?

Reply:
It looks to me that all of the *arr apps (sonarr, radarr, prowlarr, etc) are currently configured _only_ with their UIs. I have a personal TODO item to figure out how to configure them with nix. They are C# apps that either use XML or SQLite (or something else, I can't remember off the top of my head), so we'll have to figure out how to inject our configuration into a format usable by those apps.

TL;DR: no, this isn't currently configurable easily outside of the prowlarr UI.

Reply:
Then we probably shouldn't offer this option, if the port can be brought out of sync from within the webui.

Reply:
@mweinelt I disagree, for three reasons. One is consistency: the other *arr services all have a version of this option made available to the user if they want to. The second one is discoverability: as a user, if I enable the service, I have no way to know which port I should connect to when first setting-up my proxy/the software. Finally, if someone enables the option, they would know that changing the port in the webui would obviously break their setup.

Reply:
Agreed with @ambroisie on both points (discoverability and consistency).

Reply:
Not a strong opinion here, I'm a network guy by trade and coarse firewalling makes me itchy. I'm not going to use the `openFirewall` rule anyway.

Reply:
Hah, yeah I with NixOS firewalls were different. I am toying with a single nftables module in my configs instead of the builtin firewall. I understand how the builtin firewall works, but I want my rules to be really clear when I query them, and not have a ton of empty NixOS-specific chains.